### PR TITLE
feat: add tx filter endpoints

### PR DESCRIPTION
**Pull request by iamacook**
## Overview

This adds new functions/types to call the new, filterable endpoints for incoming, module and multisig transactions.

### CODE REVIEW

File: src/types/api.ts [hunk 1]
```diff
@@ -193,6 +232,77 @@ export interface operations {
...
+  /** Get filterable incoming transfers */
+  incoming_transfers: {
+    parameters: {
+      path: {
+        chainId: string
+        address: string
+      }
+      query?: {
+        execution_date__gte?: string
+        execution_date__lte?: string
```
Comment (schmanu) [inline src/types/api.ts:244]: If I compare these to the docs: https://safe.global/safe-client-gateway/docs/routes/transactions/filters/transfer/struct.TransferFilters.html

I think this should be a single underscore in the end?
```suggestion
        execution_date_gte?: string
        execution_date_lte?: string
```

File: src/types/api.ts [hunk 2]
```diff
@@ -193,6 +232,77 @@ export interface operations {
...
+  }
+  /** Get filterable module transactions */
+  module_transactions: {
+    parameters: {
+      path: {
+        chainId: string
+        address: string
+      }
+      query?: {
+        module?: string
```
Comment (schmanu) [inline src/types/api.ts:268]: In the docs: https://safe.global/safe-client-gateway/docs/routes/transactions/filters/module/struct.ModuleFilters.html

There is also a `to?: string` field.

File: src/types/api.ts [hunk 3]
```diff
@@ -193,6 +232,77 @@ export interface operations {
...
+  /** Get filterable multisig transactions */
+  multisig_transactions: {
+    parameters: {
+      path: {
+        chainId: string
+        address: string
+      }
+      query?: {
+        execution_date__gte?: string
+        execution_date__lte?: string
```
Comment (schmanu) [inline src/types/api.ts:290]: Same underscore question here:
Shouldn't it be:
```suggestion
        execution_date_gte?: string
        execution_date_lte?: string
```

**Review by schmanu — COMMENTED** [review]
Nice work adding those new routes :) 

I found some diffs between safe-client-gateway docs and the types here which need some clarification.

File: src/types/api.ts (hunk 1 above)
Comment (iamacook) [reply src/types/api.ts:244]: This is indeed confusing but they match the [transaction service parameters](https://safe-transaction.rinkeby.gnosis.io/) so the gateway can simply forward them on. We can hopefully add more of the parameters in the future and then need not adjust the name in the gateway.

File: src/types/api.ts (hunk 3 above)
Comment (iamacook) [reply src/types/api.ts:290]: See my first somment.

File: src/types/api.ts (hunk 2 above)
Comment (iamacook) [reply src/types/api.ts:268]: Nice catch!

File: src/types/api.ts (hunk 1 above)
Comment (schmanu) [reply src/types/api.ts:244]: Oh you are right.
I think this field annotation(?) in here might be responsible for that.
https://github.com/safe-global/safe-client-gateway/blob/main/src/routes/transactions/filters/transfer.rs

I didn't really know what to compare all values with :D So I just checked the docs.

**Review by schmanu — APPROVED** [review]
🚀

**Pull request merged**